pyiceberg/transforms.py

```diff
@@ -146,7 +146,13 @@ def can_transform(self, source: IcebergType) -> bool:
         return False
 
     @abstractmethod
-    def result_type(self, source: IcebergType) -> IcebergType: ...
+    def result_type(self, source: IcebergType) -> IcebergType:
+        """
+        Return the type of the result of the transform.
+
+        This type does not need to conform to the Iceberg spec, as long as it is converted to the correct type in the stored metadata.
```

Comment:
```suggestion
    This method defines both the physical and display representation of the partition field.
    
    The physical representation must conform to the Iceberg spec. The display representation
    can deviate from the spec, such as by transforming the value into a more human-readable format.
```

Reply:
@kevinzwang @corleyma wdyt? does this capture the context?

Reply:
if this is good, maybe also update the DayTranform result_type function
something like,

```
    The physical representation conforms to the Iceberg spec as DateType is internally
    converted to int. The DateType returned here provides a more human-readable way to
    display the partition field.
```

Reply:
This makes sense to me!